src/domain/runtime/type/remote_object/type/object/preview.rs

```diff
@@ -43,11 +35,11 @@ impl ObjectPreview {
     }
 }
 
-impl fmt::Display for ObjectPreview {
+impl fmt::Display for Preview {
```

Comment:
60 lines to impl Display is a lot. It might be easier to read (and more reusable) if each of the enum cases had their own display function, and when impling Display, you just matched over each enum variant and called its function.